swarm/api/http/error.go

```diff
@@ -0,0 +1,124 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+/*
+error handling for the http package
+*/
+
+package http
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/swarm/api/http/tmpl"
+)
+
+var templateMap map[int]*template.Template
+
+type ErrorParams struct {
+	Msg       string
+	Code      int
+	Timestamp string
+	template  *template.Template
+	Details   template.HTML
+}
+
+func init() {
+	initErrHandling()
+}
+
+func initErrHandling() {
+  genErrPage    := tmpl.GetGenericErrorPage()
+  notFoundPage  := tmpl.GetNotFoundErrorPage()
+	tnames := map[int]string{
+		0:   genErrPage, //default
+		400: genErrPage,
+		404: notFoundPage,
+		500: genErrPage,
+	}
+	templateMap = make(map[int]*template.Template)
+	for code, tname := range tnames {
+		templateMap[code] = template.Must(template.New(fmt.Sprint("%d",code)).Parse(tname))
+	}
```

Comment:
We should adopt the [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and just keep it as simple as possible. Today we have two templates (perhaps three), so my suggestion is pretty simple (about 10 lines of code outside the actual HTML), whereas the current code is about 30 lines of code spread across two packages, which just seems overly complex.

Always aim to solve today's problems because you never know what will happen tomorrow :wink:.